slsim/lens_pop.py

```diff
@@ -192,6 +192,40 @@ def __init__(
                 catalog_type=catalog_type,
             )
             self._source_model_type = "point_plus_extended"
+        elif source_type == "supernovae":
+            from slsim.Sources.point_sources import (
+                PointSources,
+            )
+            from slsim.Sources.SupernovaeCatalog.supernovae_sample import (
+                SupernovaeCatalog,
+            )
+
+            for key in kwargs_variability:
+                if key.startswith("ps_mag_"):
+                    suffix = key.split("ps_mag_")[1]
+            supernovae_catalog_class = SupernovaeCatalog(
+                sn_type=sn_type,
+                band=suffix,
```

Comment:
here again, might be better to hav a list of bands being passed instead of a single band